src/main/java/duke/command/UnmarkCommand.java

```diff
@@ -0,0 +1,53 @@
+package duke.command;
+
+import duke.exception.InvalidIndexException;
+import duke.tasks.TaskList;
+import duke.util.Storage;
+import duke.util.Ui;
+
+/**
+ * Represents a command to unmark a task as completed.
+ * <p>
+ * When executed, this command will unmark the task at the specified index as completed.
+ * </p>
+ */
+public class UnmarkCommand extends Command {
+
+    /** The index of the task to be unmarked as completed. */
+    private final int index;
+
+    /**
+     * Constructs a new UnmarkCommand with the specified index.
+     *
+     * @param index Index of the task to be unmarked as completed.
+     */
+    public UnmarkCommand(int index) {
+        this.index = index;
+    }
+
+    /**
+     * Executes the unmark command, unmarking the task at the specified index as completed.
+     *
+     * @param tasks List of tasks.
+     * @param ui User interface.
+     * @param storage Storage system.
+     * @return A confirmation message for unmarking of task.
+     * @throws InvalidIndexException If the given index is out of bounds.
+     */
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws InvalidIndexException {
+        String output = tasks.unmarkTask(index);
```

Comment:
Maybe we can rename 'output' to be more informative? Perhaps confimrationMessage?